sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -634,6 +638,93 @@ public void downloadToFileLeaseFail() {
         downloadFile.delete();
     }
 
+    @Test
+    public void downloadToFileWithPartialDownloads() throws Exception {
+        File uploadFile = File.createTempFile(CoreUtils.randomUuid().toString(), ".txt");
+        uploadFile.deleteOnExit();
+        Files.write(uploadFile.toPath(), DATA.getDefaultBytes());
+
+        File outFile = new File(generatePathName() + ".txt");
+        if (outFile.exists()) {
+            assertTrue(outFile.delete());
+        }
+
+        MockPartialResponsePolicy policy = new MockPartialResponsePolicy(5);
+
+        // Create a ShareFileAsyncClient for download using the custom pipeline
+        ShareFileAsyncClient downloadClient = getFileAsyncClient(ENVIRONMENT.getPrimaryAccount().getCredential(),
+            primaryFileAsyncClient.getFileUrl(), policy);
+
+        // Upload the test data
+        primaryFileAsyncClient.create(DATA.getDefaultDataSize())
+            .then(primaryFileAsyncClient.uploadFromFile(uploadFile.toString()))
+            .block();
```

Comment:
Instead of calling `block()`, we should use the `StepVerifier` for running async methods and asserting expected behavior. Here's an example of where it is done in the blob package: https://github.com/Azure/azure-sdk-for-java/blob/9f7f75dfe8eb3905d3793ec31abf34b96f7cfc2a/sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAsyncApiTests.java#L695 to match the patterns that @ibrandes has been working on updating to in our async tests. There should also be a method that we can use to assert exceptions as part of using the `StepVerifier`